apps/analytics/app/api/event/route.ts

```diff
@@ -13,36 +14,27 @@ export const runtime = "edge";
 export async function POST(request: NextRequest) {
   const event = (await request.json()) as DispatchableAnalyticsEvent;
 
-  const timestamp = JSON.stringify(event.timestamp || new Date().toISOString());
+  let generalEvent: EventInsertType = {
+    type: event.type,
+    metadata: event.metadata,
+    timestamp: event.timestamp,
+    installationId: event.installationId,
+    isocode: event.geolocation?.countryCode,
+  };
```

Comment:
This should use the types from the analytics package! If these are needed for validation, consider updating the analytics types to be derived from a zod schema.

Reply:
Here, `EventInsertType` is the type of data that we need to insert into the database, I can't use types from the analytics package here because the database schema doesn't match the analytics event type as it can be either Error or Event.
Please talk to me on Slack in case this doesn't make sense because I might have not properly expressed my thought because of my English